docs-src/docs/guides/walkthrough.md

```diff
@@ -0,0 +1,317 @@
+
+
+---
+
+### Cleanup
+
+If you're finished with the example and don't want to continue with this walkthrough, you can instruct Layer0 to delete the environment and terminate the application and the Layer0 the application was using.
+
+`l0 environment delete demo-env`
+
+However, if you intend to continue through [Deployment 2](#deployment-2-guestbook-redis), you will want to keep the resources you made in this section.
+
+
+---
+
+## 1b: Deploy with Terraform
+
+Instead of using the Layer0 CLI directly, you can instead use the Layer0 Provider for Terraform, and deploy using Terraform. 
+
+
+---
+
+### Part 1: Download the Configuration Files
+
+- [Guestbook.Dockerrun.aws.json](https://github.com/quintilesims/layer0-examples/blob/guides-overhaul/iterative-walkthrough/guestbook/Guestbook.Dockerrun.aws.json)
+- [terraform.tfvars](https://github.com/quintilesims/layer0-examples/blob/guides-overhaul/iterative-walkthrough/terraform/terraform.tfvars)
+- [layer0.tf](https://github.com/quintilesims/layer0-examples/blob/guides-overhaul/iterative-walkthrough/terraform/deployment-1/layer0.tf) for Deployment 1
+
+
+---
+
+### Part 2: Terraform Apply
+
+Run `terraform apply` to begin the process. Terraform will prompt you for configuration values that it does not have.
+
+To begin deploying the application, run the following command:
+
+`terraform apply`
+
+_To avoid entering these values manually each time you run terraform, you can set the terraform variables by editing the `terraform.tfvars` file._
+
+```
+var.endpoint
+  Enter a value: <enter your Layer0 endpoint>
+
+var.token
+  Enter a value: <enter your Layer0 token>
+
+layer0_environment.demo: Refreshing state...
+...
+...
+...
+layer0_service.guestbook: Creation complete
+
+Apply complete! Resources: 7 added, 0 changed, 0 destroyed.
+
+The state of your infrastructure has been saved to the path
+below. This state is required to modify and destroy your
+infrastructure, so keep it safe. To inspect the complete state
+use the `terraform show` command.
+
+State path: terraform.tfstate
+
+Outputs:
+
+guestbook_url = <http endpoint for the sample application>
+```
+
+__It may take a few minutes for the guestbook service to launch and the load balancer to become available. During that time you may get HTTP 503 errors when making HTTP requests against the load balancer URL.__
+
+Terraform will set up the entire environment for you and then output a link to the application's load balancer.
+
+
+### What's happening
+
+Terraform provisions the AWS resources (an RDS instance, VPC and subnet configurations to connect the RDS instance to the Layer0 application), configures environment variables for the application, and deploys the application into a Layer0 environment.
+
+You can use Terraform with Layer0 and AWS to create "fire-and-forget" deployments for your applications.
```

Comment:
This sentence I would expect at the start of 1b. An intro into why we would use Terraform.